### PR TITLE
Cache cell volumes.

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

File: libecl/src/ecl_grid.c [hunk 1]
```diff
@@ -1230,58 +1231,65 @@ static inline double tetrahedron_volume( tetrahedron_type tet ) {
...
-          tet.p3 = corners[ point2 ];
-          volume += tetrahedron_volume( tet ) / 6;
+	const int point0 = tetrahedron_permutations[ method ][ itet ][ 0 ];
+	const int point1 = tetrahedron_permutations[ method ][ itet ][ 1 ];
+	const int point2 = tetrahedron_permutations[ method ][ itet ][ 2 ];
+	
+	tet.p1 = corners[ point0 ];
+	tet.p2 = corners[ point1 ];
+	tet.p3 = corners[ point2 ];
+	volume += tetrahedron_volume( tet ) / 6;
```
Comment (pgdr) [inline libecl/src/ecl_grid.c]: And out of nowhere tabs appeared!

Is there a reason you do not `/6` after the looping?

File: libecl/src/ecl_grid.c [hunk 2]
```diff
@@ -694,6 +694,7 @@ static void point_copy_values(point_type * p , const point_type * src) {
 #define CELL_FLAG_TAINTED  4     /* lazy fucking stupid reservoir engineers make invalid grid
                                     cells - for kicks??  must try to keep those cells out of
                                     real-world calculations with some hysteric heuristics.*/
+#define CELL_FLAG_VOLUME 8
```
Comment (pgdr) [inline libecl/src/ecl_grid.c]: Would it be possible to initialize volume with -1 and skip this flag?

File: libecl/src/ecl_grid.c (hunk 1 above)
Comment (pgdr) [inline libecl/src/ecl_grid.c]: Perhaps we could change the name of `tetrahedron_volume` to something else if it doesn't actually compute the volume.

**Comment by pgdr** [comment]
Go ahead and merge once comments are addressed.

File: libecl/src/ecl_grid.c (hunk 2 above)
Comment (joakim-hove) [inline libecl/src/ecl_grid.c]: > Would it be possible to initialize volume with -1 and skip this flag?

There are cases (not active now but still) - where the signed volume is interesting.

**Pull request merged**